packages/storage/amplify_storage_s3_dart/lib/src/model/s3_storage_download_data_operation.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3_dart/amplify_storage_s3_dart.dart';
+
+/// {@template storage.amplify_storage_s3.download_data_operation}
+/// An operation created by calling the Storage S3 plugin `downloadData` API.
+/// {@endtemplate}
+class S3StorageDownloadDataOperation extends StorageDownloadDataOperation<
+    StorageDownloadDataRequest<S3StorageDownloadDataOptions>,
+    S3StorageDownloadDataResult> {
+  /// {@macro storage.amplify_storage_s3.download_data_operation}
+  S3StorageDownloadDataOperation({
+    required super.request,
+    required super.result,
+    required Future<void> Function() resume,
+    required Future<void> Function() pause,
+    required Future<void> Function() cancel,
+  })  : _resume = resume,
+        _pause = pause,
+        _cancel = cancel;
+
+  final Future<void> Function() _resume;
+  final Future<void> Function() _pause;
+  final Future<void> Function() _cancel;
+
+  /// Resumes the operation that is in a paused state.
```

Comment:
These should be on the base class.

Reply:
The base class is abstract and requires implements these methods, e.g. 

```dart
abstract class StorageDownloadDataOperation<
        Request extends StorageDownloadDataRequest,
        Result extends StorageDownloadDataResult>
    extends StorageOperation<Request, Result>
    implements StorageControllableOperation {
  StorageDownloadDataOperation({
    required super.request,
    required super.result,
  });
}
```

Reply:
I mean the doc comments should be on the StorageControllableOperation since they're very generic and don't need to be repeated